ec/src/hashing/curve_maps/swu/mod.rs

```diff
@@ -0,0 +1,181 @@
+use core::marker::PhantomData;
+
+use ark_ff::bytes::{ToBytes};
+
+use crate::models::SWModelParameters;
+use ark_ff::{vec::Vec, Field, One, SquareRootField, Zero};
+use ark_std::string::ToString;
+
+use crate::{
+    hashing::{map_to_curve_hasher::MapToCurve, HashToCurveError},
+    models::short_weierstrass_jacobian::GroupAffine,
+    AffineCurve,
+};
+
+/// Trait defining the necessary parameters for the SWU hash-to-curve method
+/// for the curves of Weierstrass form of:
+/// y^2 = x^3 + a*x + b where ab != 0. From [WB2019]
+///
+/// - [WB19] Wahby, R. S., & Boneh, D. (2019). Fast and simple constant-time
+///   hashing to the bls12-381 elliptic curve. IACR Transactions on
+///   Cryptographic Hardware and Embedded Systems, nil(nil), 154–179.
+///   http://dx.doi.org/10.46586/tches.v2019.i4.154-179
+pub trait SWUParams: SWModelParameters {
+    // we need an element of the base field which is not a square root see [1] Sect.
+    // 4. it is also convenient to have $g(b/xi * a)$ to be square. In general
+    // we use a xi with low absolute value coefficients when they are
+    // represented as element of ZZ.
+    const XI: Self::BaseField; // a nonsquare in Fq
+    const ZETA: Self::BaseField; // arbitatry root of unity
+    const XI_ON_ZETA_SQRT: Self::BaseField; // square root of THETA
+}
+
+pub struct SWUMap<P: SWUParams> {
+    pub domain: Vec<u8>,
+    curve_params: PhantomData<fn() -> P>,
+}
+
+impl<P: SWUParams> MapToCurve<GroupAffine<P>> for SWUMap<P> {
+    /// This is to verify if the provided SWUparams makes sense, doesn't do much
+    /// for now
+    fn new_map_to_curve(domain: &[u8]) -> Result<Self, HashToCurveError> {
+        // Verifying that both XI and ZETA are non-squares
+        if P::XI.legendre().is_qr() || P::ZETA.legendre().is_qr() {
+            return Err(HashToCurveError::MapToCurveError(
+                "both Xi and Zeta should be quadratic non-residues for the SWU map".to_string(),
```

Comment:
```suggestion
                "both xi and zeta should be quadratic non-residues for the SWU map".to_string(),
```
(lowercase ξ and ζ are different to uppercase Ξ and Ζ).